test/unit/services/session_store_test.rb

```diff
@@ -43,4 +43,19 @@ def session_store
       assert_equal [:a, node_name, :c], session[flow_name].keys
     end
   end
+
+  context "#clear" do
+    should "remove entries from session" do
+      assert_not_equal({}, session)
```

Comment:
Why are we making an assertion before we've done the action?

Reply:
The hash being empty is the default behaviour. So checking that the hash is empty after it is cleared, could mean it has been cleared, but it could also mean that it was empty to start with and nothing changed. So checking the hash is not empty beforehand makes sure the clearing is actually doing something. I was hoping to use `assert_changes` but that didn't seem to work.

I'd add, the test this code replaces is an example of the problem of using just a test of emptiness. The test passed because the session was empty at the end of the test. The problem was that session was always empty, so in fact the test wasn't testing the intended behaviour.

Reply:
Why can we not set the session not be empty, rather asserting it? So we no for certain the we are calling "#clear" on sessions that has data?

Reply:
I've changed the assert to a specific call that ensures the session isn't empty